ui/guest/src/components/Guest.js

```diff
@@ -978,8 +1004,76 @@ export function Guest(props) {
           highlighted: {}
         }
       });
-    }
+    },
+
+    dragenter(e, record) {
+      if (stateRef.current.common.status === EditingStatus.LISTENING && e.originalEvent.dataTransfer.types.includes('Files')) {
```

Comment:
I think dataTransfer or types may be null/undefined and throw.

`e.originalEvent.dataTransfer?.types.?includes('...`

Break the if conditions each on it's own line.